web/relative.go

```diff
@@ -29,8 +29,18 @@ func RelativeEndpoint(request *http.Request, endpoint string) string {
 		host = "localhost"
 	}
 
-	resultUrl := url.URL{Scheme: proto, Host: host, Path: endpoint}
-	result = resultUrl.String()
+	// Build query parameters.
+	query := url.Values{}
+	for key, value := range queryParams {
+		query.Add(key, value)
+	}
```

Comment:
This will iterate over the queryParams in random order. I _think_ that's a good thing -- if we were to ever add a second query param, we don't want people assuming anything about the order of them -- but I want to make sure we're in agreement on that.

Reply:
I can't really think of a precedent for query params needing to have a specific order, can you?